### ISSUE TITLE
TFDS dataset generation fails for qcd_high_pt in gen-jet clustering

### ISSUE THREAD

**Issue by jpata** [jpata/particleflow#133]
This tfds generation failed at some point halfway through.

```
tfds build hep_tfds/heptfds/cms_pf/qcd_high_pt
...
Traceback (most recent call last):                              
  File "/usr/local/bin/tfds", line 8, in <module>                       
    sys.exit(launch_cli())
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/scripts/cli/main.py", line 102, in launch_cli
    app.run(main, flags_parser=_parse_flags)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/scripts/cli/main.py", line 97, in main
    args.subparser_fn(args)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/scripts/cli/build.py", line 192, in _build_datasets
    _download_and_prepare(args, builder)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/scripts/cli/build.py", line 342, in _download_and_prepare
    builder.download_and_prepare(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/core/dataset_builder.py", line 481, in download_and_prepare
    self._download_and_prepare(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/core/dataset_builder.py", line 1218, in _download_and_prepare
    future = split_builder.submit_split_generation(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/core/split_builder.py", line 310, in submit_split_generation
    return self._build_from_generator(**build_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow_datasets/core/split_builder.py", line 371, in _build_from_generator
    for key, example in utils.tqdm(
  File "/usr/local/lib/python3.8/dist-packages/tqdm/std.py", line 1195, in __iter__
    for obj in iterable:
  File "/home/joosep/particleflow/hep_tfds/heptfds/cms_utils.py", line 191, in generate_examples
    X, ygen, ycand = prepare_data_cms(str(fi), pad_size)
  File "/home/joosep/particleflow/hep_tfds/heptfds/cms_utils.py", line 162, in prepare_data_cms
    jet_constituents = [index_mapping[idx] for idx in constituent_idx[jet_idx]] # map back to constituent index *before* masking
  File "/home/joosep/particleflow/hep_tfds/heptfds/cms_utils.py", line 162, in <listcomp>
    jet_constituents = [index_mapping[idx] for idx in constituent_idx[jet_idx]] # map back to constituent index *before* masking
IndexError: index 4659 is out of bounds for axis 0 with size 4659
```

The other ones (ttbar, ztt, qcd) succeeded, so I'm debugging this a bit.

**Comment by jpata** [comment]
Traced further to the file `QCD_Pt_3000_7000_14TeV_TuneCUETP8M1_cfi/raw/pfntuple_3392.pkl.bz2`

**Comment by jmduarte** [comment]
@jpata thanks for tracing it to that file, it helped a lot. i think the bug fix is here: https://github.com/jpata/hep_tfds/pull/5

**Comment by jpata** [comment]
Thanks a lot! The bump of hep_tfds is included in #132 - let's see if it runs through for all datasets (should be done in about 24h).

**Comment by jpata** [comment]
confirmed that it doesn't crash after https://github.com/jpata/hep_tfds/pull/5 on any event in the dataset, so it's fixed by #132.
